src/htm/engine/RESTapi.cpp

```diff
@@ -36,44 +36,43 @@ RESTapi::~RESTapi() { }
 
 RESTapi* RESTapi::getInstance() { return &rest; }
 
-std::string RESTapi::get_new_id_(const std::string &old_id) {
-  std::string id = old_id;
-
-  // re-use the same id if we can or use a provide one.
+  std::string RESTapi::get_new_id_() {
+  // No id was provided so find the next available number.
+  // Do not use an id of "0".
+  
   std::map<std::string, ResourceContext>::iterator itr;
-  if (id.empty()) {
-    // id is empty, create a new context id
-    while (rest.resource_.size() < UINT16_MAX - 1) {
-      unsigned int id_nbr = next_id++;
-      if (id_nbr == 0)
-        id_nbr = next_id++; // allow integer wrap of the id without a 0 value.
-      char buf[10];
-      std::snprintf(buf, sizeof(buf), "%4.04x", id_nbr);
-      id = buf;
-
-      // Make sure this new session id is not in use.
-      itr = resource_.find(id);
-      if (itr == resource_.end() || (itr->second.t < time(0) - RESOURCE_TIMEOUT)) {
-        // This is one we can use
-        break;
-      }
-    };
+  std::string id;
+  while (rest.resource_.size() < UINT16_MAX - 1) {
```

Comment:
what is this while doing? Seems it's finding a freee id to reuse (?)

Reply:
> Seems it's finding a free id to reuse (?)

Yes, it is possible to have both generated ids (from a previous wrap) and id's provided by the user that could already exist. So when generating a new id we need to make sure it is not a duplicate.

Reply:
ok as is. a nit: didn't I read somewhere that the ID is in [1, 9999], so it might be nice to make a `size_t ID_MAX=9999` and use that in the loop. will make it code clearer to read.